Ubika/ubika_modules/connector_ubika_cloud_protector_alerts.py

```diff
@@ -0,0 +1,229 @@
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._stop_event = Event()
+        self.context = PersistentJSON("context.json", self._data_path)
+        self.from_date = self.most_recent_date_seen
+
+        # Register signal to terminate thread
+        signal.signal(signal.SIGINT, self.exit)
+        signal.signal(signal.SIGTERM, self.exit)
+
+    def exit(self, _, __) -> None:
+        self.log(message="Stopping Ubika Cloud Protector connector", level="info")  # pragma: no cover
+        # Exit signal received, asking the processor to stop
+        self._stop_event.set()
+
+    @property
+    def most_recent_date_seen(self) -> datetime:
+        now = datetime.now(timezone.utc)
+
+        with self.context as cache:
+            most_recent_date_seen_str = cache.get("most_recent_date_seen")
+
+        # if undefined, retrieve events from the last hour
+        if most_recent_date_seen_str is None:
+            return now - timedelta(hours=1)
+
+        # parse the most recent date seen
+        most_recent_date_seen = isoparse(most_recent_date_seen_str)
+
+        # we don't retrieve messages older than one week
+        one_week_ago = now - timedelta(days=7)
+        if most_recent_date_seen < one_week_ago:
+            most_recent_date_seen = one_week_ago
+
+        return most_recent_date_seen
+
+    @cached_property
+    def client(self) -> ApiClient:
+        return ApiClient(
+            self.configuration.token,
+        )
+
+    def _handle_response_error(self, response: requests.Response) -> None:
+        if not response.ok:
+            message = (
+                f"Request on Ubika Cloud Protector API to fetch events failed with status "
+                f"{response.status_code} - {response.reason}"
+            )
+
+            try:
+                error = response.json()
+                message = f"{message}: {error['errorCode']} - {error['errorSummary']}"
+
+            except Exception:
+                pass
+
+            raise FetchEventsException(message)
+
+    def __fetch_next_events(self, from_date: datetime) -> Generator[list, None, None]:
+        # get the first page of events
+        headers = {"Content-Type": "application/json"}
+        url = f"{self.BASE_URI}/providers/{self.configuration.provider}/tenants/{self.configuration.tenant}/alertlogs"
+        params = {
+            "start_time": int(from_date.timestamp()),
+            "limit": self.configuration.chunk_size,
+        }
+
+        response = self.client.get(url, params=params, headers=headers, timeout=60)
+
+        while self.running:
+            # manage the last response
+            self._handle_response_error(response)
+
+            # get events from the response
+            events = response.json()
+
+            # yielding events if defined
+            if events and len(events["items"]) > 0:
+                INCOMING_MESSAGES.labels(intake_key=self.configuration.intake_key).inc(len(events))
```

Comment:
I think we want to add the number of alerts in the response.
```suggestion
                INCOMING_MESSAGES.labels(intake_key=self.configuration.intake_key).inc(len(events["items"]))
```